rust/libnewsboat/src/utils.rs

```diff
@@ -477,6 +479,19 @@ pub fn getcwd() -> Result<PathBuf, io::Error> {
     env::current_dir()
 }
 
+/// Recursively create directories if missing and set permissions accordingly.
+pub fn mkdir_parents(p: &str, mode: u32) -> isize {
+    let path = Path::new(p);
```

Comment:
I think it's better to accept `&Path` in Rust, and let the FFI bindings convert from `const char*` to that. This way, our C++ code keeps working and our Rust code is forced to use a more appropriate `Path` type.